### DISCUSSION TITLE
myMpd and mpd socket

### DISCUSSION THREAD

**Discussion by antonellocaroli** [jcorporation/myMPD#561] [Q&A] [answered]
how come mympd as default setting has as host mpd /run/mpd/socket

it should not be /var/run/mpd/socket

as recommended also in the documentation?

https://mpd.readthedocs.io/en/stable/user.html#configuration

`To bind to a local socket (UNIX domain socket), specify an absolute path or a path starting with a tilde (~). Some clients default to connecting to /var/run/mpd/socket so this may be a good choice:`

**Comment by jcorporation** [comment, marked as answer]
Nowadays most distributions defaults to `/run/mpd/socket`. It seems for me the more modern approach.

**Reply by GK-774** [reply]
Yes most current distributions default to /run instead of the older /var/run and usually have a /var/run as link to /run for backwards compatibility. So either way will work but /run/mpd/socket is future proof.

**Comment by antonellocaroli** [comment]
Ok, the majority of distributions use /run/mpd/socket, but mine uses /var/run/mpd/socket. 
This is not a problem....but the first time it tries to change settings mympd quickly updates itself to the default speed....I have to try several times to get my settings to work. Personally I preferred when there was a configuration file where you manually set some configurations, I found it more functional...

**Comment by jcorporation** [comment]
myMPD tries both standard locations with commit 9565b2fe83432a33d83316484211c8ea3503754b. 

You can set the mpd connection details through environment at first startup: https://jcorporation.github.io/myMPD/configuration/

But changing the settings in the connection dialog should be straight forward.

A configuration file and configureable settings in the ui had lead to confusion in the past. That’s why the config file was removed.

**Comment by nikonakoneko** [comment]
Note that on gentoo the default mpd config use /var/lib/mpd/socket (is that upstream configuration, maybe?)

**Comment by jcorporation** [comment]
I do not think that comes from upstream, the mpd manual references /run/mpd/socket.

https://mpd.readthedocs.io/en/latest/client.html#environment-variables

Regardless, I can add this path as another path to test for the mpd socket.
